File: test/net/azib/photos/GalleryLoaderTest.java
package net.azib.photos;

import org.junit.Test;

import static org.hamcrest.CoreMatchers.is;
import static org.junit.Assert.assertThat;

public class GalleryLoaderTest {
  @Test
  public void parse() throws Exception {
    Gallery gallery = new XMLParser<>(new GalleryLoader()).parse(getClass().getResourceAsStream("gallery.xml"));
    assertThat(gallery.getAuthor(), is("Anton Keks"));
    assertThat(gallery.getTimestamp(), is(1464106391641L));
    assertThat(gallery.getAlbums().size(), is(1));

    Album album = gallery.getAlbums().get(0);
    assertThat(album.getName(), is("Chernobyl"));
    assertThat(album.getTitle(), is("Chernobyl"));
    assertThat(album.getDescription(), is("Apocalyptic experience in Chernobyl and Pripyat, a soviet city abandoned in 1986 after the nuclear disaster. Current radiation levels are compatible with life :-)"));
    assertThat(album.getAuthor(), is("Anton Keks"));
    assertThat(album.isPublic(), is(true));
    assertThat(album.getThumbUrl(), is("https://lh3.googleusercontent.com/-EfV7Xxjk3gk/VjfV9bujtGE/AAAAAAABKUY/gQBUlooE9lsYdyZ1O7ciOiGo-5pch3_DQCHM/s212-c/Chernobyl"));
    assertThat(album.getTimestamp(), is(1432450800000L));
    assertThat(album.getGeo().getLat(), is(51.276303f));
    assertThat(album.getGeo().getLon(), is(30.221899f));
    assertThat(album.size(), is(159));
  }
}

File: test/net/azib/photos/GalleryLoaderTest.kt
package net.azib.photos

import org.assertj.core.api.Assertions.assertThat
import org.jetbrains.spek.api.Spek

class GalleryLoaderTest: Spek({
  val xml = Gallery::class.java.getResourceAsStream("gallery.xml")

  it("parses gallery feed") {
    val gallery = XMLParser(GalleryLoader(212)).parse(xml)
    assertThat(gallery.authorId).isEqualTo("117440562642491680332")
    assertThat(gallery.author).isEqualTo("Anton Keks")
    assertThat(gallery.timestampISO).isEqualTo("2016-05-24T19:13:11Z")
    assertThat(gallery.albums.size).isEqualTo(1)

    val album = gallery.albums.values.first()
    assertThat(album.id).isEqualTo("6212669462372660321")
    assertThat(album.name).isEqualTo("Chernobyl")
    assertThat(album.title).isEqualTo("Chernobyl")
    assertThat(album.description).isEqualTo("Apocalyptic experience in Chernobyl and Pripyat, a soviet city abandoned in 1986 after the nuclear disaster. Current radiation levels are compatible with life :-)")
    assertThat(album.author).isEqualTo("Anton Keks")
    assertThat(album.isPublic).isEqualTo(true)
    assertThat(album.thumbUrl).isEqualTo("https://lh3.googleusercontent.com/-EfV7Xxjk3gk/VjfV9bujtGE/AAAAAAABKUY/gQBUlooE9lsYdyZ1O7ciOiGo-5pch3_DQCHM/s212-c/Chernobyl.jpg")
    assertThat(album.timestampISO).isEqualTo("2015-11-05T21:37:39Z")
    assertThat(album.geo!!.lat).isEqualTo(51.276303f)
    assertThat(album.geo!!.lon).isEqualTo(30.221899f)
    assertThat(album.size()).isEqualTo(159)
  }
})